src/workspace-data/data-table/entity-service/EntityDeleter.js

```diff
@@ -27,11 +27,20 @@ export const EntityDeleter = ({ onDismiss, onSuccess, namespace, name, selectedE
     } catch (error) {
       switch (error.status) {
         case 409:
-          setAdditionalDeletions(_.filter((entity) => entity.entityType !== selectedDataType, await error.json()));
+          setAdditionalDeletions(
+            _.filter(
+              (errorEntity) =>
+                !_.some(
+                  (selectedEntity) => selectedEntity.entityType === errorEntity.entityType && selectedEntity.entityName === errorEntity.entityName,
+                  entitiesToDelete
+                ),
+              await error.json()
+            )
+          );
```

Comment:
Optional, I won't block the PR on this feedback: the logic here to "find all entities in the json response which were not originally selected by the user" is somewhat complex. You could factor this out into a standalone function and write unit tests for it. I'm ok shipping as-is though.